### PR TITLE
Add basecamp contract with both mars and xmars tokens

### PR DESCRIPTION
**Pull request by spikeonmars**

### CODE REVIEW

File: contracts/basecamp/src/contract.rs [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+                            msg: to_binary(&HandleMsg::InitTokenCallback { token_id: 0 }).unwrap(),
+                            contract_addr: HumanAddr::from(MOCK_CONTRACT_ADDR),
+                        }),
+                    })
+                    .unwrap(),
+                    send: vec![],
+                    label: None,
+                }),
+                CosmosMsg::Wasm(WasmMsg::Instantiate {
+                    code_id: 11,
```
Comment (ayushmenon) [inline contracts/basecamp/src/contract.rs:222]: should this also be 11u64 for consistency?

File: contracts/basecamp/src/state.rs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+use cosmwasm_storage::{singleton, singleton_read, ReadonlySingleton, Singleton};
+
+// keys (for singleton)
+pub static CONFIG_KEY: &[u8] = b"config";
+
+// namespaces (for buckets)
+pub static RESERVES_NAMESPACE: &[u8] = b"reserves";
+pub static DEBTS_NAMESPACE: &[u8] = b"debts";
+pub static USERS_NAMESPACE: &[u8] = b"users";
+pub static RESERVE_DENOMS_NAMESPACE: &[u8] = b"reserve_denoms";
```
Comment (ayushmenon) [inline contracts/basecamp/src/state.rs:14]: will we require all of these?

**Review by ayushmenon — COMMENTED** [review]
looks great! Just left two minor comments / questions

File: contracts/basecamp/src/contract.rs (hunk 1 above)
Comment (spikeonmars) [reply contracts/basecamp/src/contract.rs:222]: yep good catch. Actually the others should be just 11 as the compiler should know the type

File: contracts/basecamp/src/state.rs (hunk 2 above)
Comment (spikeonmars) [reply contracts/basecamp/src/state.rs:14]: nope, will delete

**Pull request merged**